spec/support/shared_examples/api_index_endpoint_support.rb

```diff
@@ -51,6 +51,24 @@
   end
 end
 
+RSpec.shared_examples "an API index endpoint on a parent resource" do
+  context "when 2 resources exist for current_lead_provider" do
+    let!(:resource) { create_resource(lead_provider: current_lead_provider) }
+
+    before do
+      create_resource_with_different_parent(lead_provider: current_lead_provider)
+    end
+
+    it "only returns the resource nested on the requested parent" do
```

Comment:
This test seems too abstract. Its hard to know what we are testing.
Maybe this would be better as an outcome specific shared example so that the test names can be specific and clear.

Reply:
I've pushed an update to pass in the child/parent resource names from the outer spec, so it now reads a bit better:

>behaves like an API index endpoint on a parent resource
      when 2 outcomes exist for current_lead_provider
        only returns the outcome nested on the requested participant